auth/src/main/java/io/apicurio/common/apps/auth/authn/AppAuthenticationMechanism.java

```diff
@@ -222,17 +236,31 @@ public String getForwardedFor() {
 
     @Override
     public Uni<ChallengeData> getChallenge(RoutingContext context) {
-        return oidcAuthenticationMechanism.getChallenge(context);
+        var enabledAuth = selectEnabledAuth();
+        if (enabledAuth != null) {
+            return enabledAuth.getChallenge(context);
+        } else {
+            return Uni.createFrom().nullItem();
+        }
     }
 
     @Override
     public Set<Class<? extends AuthenticationRequest>> getCredentialTypes() {
-        return Collections.singleton(TokenAuthenticationRequest.class);
+        Set<Class<? extends AuthenticationRequest>> credentialTypes = new HashSet<>();
+        credentialTypes.addAll(oidcAuthenticationMechanism.getCredentialTypes());
+        credentialTypes.addAll(basicAuthenticationMechanism.getCredentialTypes());
+        return credentialTypes;
     }
 
     @Override
-    public HttpCredentialTransport getCredentialTransport() {
-        return new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
+    public Uni<HttpCredentialTransport> getCredentialTransport(RoutingContext context) {
+        var enabledAuth = selectEnabledAuth();
+        if (enabledAuth != null) {
+            // TODO: was new HttpCredentialTransport(HttpCredentialTransport.Type.AUTHORIZATION, "bearer");
```

Comment:
Should this be changed to honour the configured auth type?

Reply:
it is, I just need to remove the TODO comment which is a leftover

Reply:
done

Reply:
oh right, it's using the credential transport from the enabled auth